pkg/pkg_test.go

```diff
@@ -6,34 +6,115 @@ import (
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 	"github.com/openshift-online/ocm-csv-parser/pkg/convert"
+	"github.com/openshift-online/ocm-csv-parser/pkg/errormsgs"
+	. "github.com/openshift-online/ocm-csv-parser/pkg/fieldnames"
+	"github.com/openshift-online/ocm-csv-parser/pkg/validations"
+	errors "github.com/zgalor/weberr"
+)
+
+const (
+	testOutputFile = `---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cloud-resources-config
+data:
+  instance-types.yaml: |
+    instance_types:
+    - id: m7a.12xlarge
+      name: m7a.12xlarge - General purpose
+      cloud_provider_id: aws
+      cpu_cores: 48
+      memory: 206158430208
+      category: general_purpose
+      size: 12xlarge
+      generic_name: standard-48-m7a
+`
 )
 
 var _ = Describe("Pkg", func() {
-	Context("Functions should return the correct output", func() {
+	Context("Functions should return the correct output", Ordered, func() {
 		var machineTypeId string
 		var machineTypes []map[string]interface{}
 		var outputFile string
 		BeforeEach(func() {
 			machineTypeId = "m7a.12xlarge"
 			outputFile = "test/test.configmap.yaml"
 		})
+		AfterAll(func() {
+			os.Remove("test/test.configmap.yaml")
+		})
 		It("Convert single resource from CSV into YAML", func() {
 			var err error
 			machineTypes, err = convert.CsvToResources("test/test.csv")
-			Expect(err).To(BeNil())
+			Expect(err).ToNot(HaveOccurred())
 			machineType := machineTypes[0]
-			Expect(machineType["id"]).To(Equal(machineTypeId))
-			Expect(machineType["size_pretty"]).To(Equal("12xlarge"))
-			Expect(machineType["cpu_cores"]).To(Equal("48"))
+			Expect(machineType[Id]).To(Equal(machineTypeId))
+			Expect(machineType[SizePretty]).To(Equal("12xlarge"))
+			Expect(machineType[CpuCores]).To(Equal("48"))
 		})
 		It("Convert parsed resource into yaml", func() {
 			err := convert.ResourcesToYamlMachineTypes(machineTypes, outputFile)
-			Expect(err).To(BeNil())
+			Expect(err).ToNot(HaveOccurred())
 			testFile, err := os.ReadFile(outputFile)
 
 			Expect(testFile).To(ContainSubstring(machineTypeId))
```

Comment:
`err` is assigned, but never tested.
I believe we should test that no error has occurred:
```suggestion
			testFile, err := os.ReadFile(outputFile)
			Expect(err).ToNot(HaveOccurred())
			Expect(testFile).To(ContainSubstring(machineTypeId))
```